internal/pkg/agent/application/upgrade/artifact/config.go

```diff
@@ -157,10 +164,12 @@ func DefaultConfig() *Config {
 	transport.Timeout = 120 * time.Minute
 
 	return &Config{
-		SourceURI:             DefaultSourceURI,
-		TargetDirectory:       paths.Downloads(),
-		InstallPath:           paths.Install(),
-		HTTPTransportSettings: transport,
+		SourceURI:              DefaultSourceURI,
+		TargetDirectory:        paths.Downloads(),
+		InstallPath:            paths.Install(),
+		RetryMaxCount:          6,
```

Comment:
Thoughts on increasing the default number of retries? We used 6 before, not sure there was a particular reason.

I almost thing we should just rely exclusively on the default timeout, we retry for up to 2 hours regardless of how many attempts that works out to be.

Reply:
> We used 6 before, not sure there was a particular reason.

Yeah, I didn't know the history/rationale there either so I just carried it over.

> I almost thing we should just rely exclusively on the default timeout, we retry for up to 2 hours regardless of how many attempts that works out to be.

Yeah, this makes sense to me, especially with the randomized nature of the backoff durations between retries.  Will remove the `RetryMaxCount` setting from this PR.

Reply:
Removed `RetryMaxCount` in 56f336a24.